LocalMultiplayerAgent/DeploymentTool/DeploymentScript.cs

```diff
@@ -0,0 +1,496 @@
+        }
+
+        public async Task CheckAndUploadCertificatesAsync(string certSkipToken = null, int pageSize = 10)
+        {
+            List<string> failedCertUploads = new List<string>();
+
+            do
+            {
+                var certificateSummariesRequest = new ListCertificateSummariesRequest
+                {
+                    SkipToken = certSkipToken,
+                    PageSize = pageSize
+                };
+                var certificateSummariesResponse = await PlayFabMultiplayerAPI.ListCertificateSummariesAsync(certificateSummariesRequest);
+
+                ErrorCheck(certificateSummariesResponse.Error);
+
+                certSkipToken = certificateSummariesResponse.Result.SkipToken ?? null;
+
+                foreach (var certificate in settings.GameCertificateDetails)
+                {
+                    IEnumerable<CertificateSummary> existingCerts = certificateSummariesResponse.Result.CertificateSummaries.Where(x => x.Name == certificate.Name);
+                    if (!existingCerts.Any() && string.IsNullOrEmpty(certSkipToken))
+                    {
+                        var uploadCertificateRes = await UploadCertificateAsync(certificate);
+
+                        if (uploadCertificateRes.Error != null)
+                        {
+                            Console.WriteLine(uploadCertificateRes.Error.ErrorMessage);
+                            failedCertUploads.Add(certificate.Name);
+                            continue;
+                        }
+
+                        else
+                        {
+                            Console.WriteLine($"Uploading {certificate.Name} successful!");
+                        }
+                    }
+                }
+
+            } while (!string.IsNullOrEmpty(certSkipToken));
+
+            if (failedCertUploads?.Count > 0)
+            {
+                Console.WriteLine($"The folowing certificates failed in the upload process: {string.Join(", ", failedCertUploads)}");
+            }
+        }
+
+        public List<PlayFab.MultiplayerModels.Port> PortMapping()
+        {
+            var ports = new List<PlayFab.MultiplayerModels.Port>();
+
+            foreach (var portList in settings.PortMappingsList)
+            {
+                ports.AddRange(portList?.Select(x => new PlayFab.MultiplayerModels.Port()
+                {
+                    Name = x.GamePort.Name,
+                    Num = settings.RunContainer ? x.GamePort.Number : 0,
+                    Protocol = Enum.Parse<ProtocolType>(x.GamePort.Protocol)
+                }).ToList());
+            }
+
+            return ports;
+        }
+
+        public async Task UploadAssetFileAsync(string uploadUrl, string fileNamePath)
+        {
+            string fileName = Path.GetFileName(fileNamePath);
+            try
+            {
+                //TODO: log progress of asset upload
```

Comment:
for TODOs on open source projects, it's a good practice to create an issue on the repository and link it here